### ISSUE TITLE
数据集投毒

### ISSUE THREAD

**Issue by clearloveclearlove** [vtu81/backdoor-toolbox#3]
朋友，在 create_poisoned_set.py 进行数据集投毒时，怎么直接从所有样本中选择一定比例的样本进行投毒，我的理解是不应该只从非target class中的样本选吗

**Comment by vtu81** [comment]
Hi,

Thanks for your question about poison samples selection during the poisoning process.

In our implementation, given a `poison_rate`, we randomly choose a corresponding number of training samples from **all classes** to inject backdoor trigger -- this is a reasonable setting, since sometimes an adversary would simply inject the backdoor triggers into the training set without knowing the training samples' ground-truth labels. We also understand that in some other threat models, the backdoor adversary may attempt to avoid selecting training samples from the target class (i.e. only from non-target class), which might somehow increase the poisoning efficacy. But according to our empirical observations, poisoning samples from 1) *all classes* and 2) only *non-target classes* are not very much different (e.g. the ASR are similar). Therefore, we simply select poison samples without considering their ground-truth labels, as this is more straightforward and common.

Of course, you are more than welcome to alter the corresponding scripts in [poison_tool_box](poison_tool_box), if you would like to further study different poisoning strategies (e.g. selecting poisoning samples from non-target class only).

Best,
